File: examples/pyspark/sparkapp.py
from pyspark import SparkContext, SparkConf
from pyspark.sql import SQLContext, SparkSession
from pyspark import sql

# Create the spark session
spark = SparkSession \
    .builder \
    .appName("Vertica Connector Pyspark Example") \
    .getOrCreate()
spark_context = spark.sparkContext
sql_context = sql.SQLContext(spark_context)

# The name of our connector for Spark to look up
format = "com.vertica.spark.datasource.VerticaSource"

# Set connector options based on our Docker setup
host="vertica"
user="dbadmin"
password=""
db="docker"
staging_fs_url="webhdfs://hdfs:50070/data/"
table="pysparktest"

# Define data to write to Vertica
columns = ["language","users_count"]
data = [("Java", "20000"), ("Python", "100000"), ("Scala", "3000")]
# Create an RDD from the data
rdd = spark_context.parallelize(data)
# Convert the RDD to a DataFrame
df = rdd.toDF(columns)
# Write the DataFrame to the Vertica table pysparktest
df.write.mode('overwrite').save(
 # Spark format
 format=format,
 # Connector specific options
 host=host,
 user=user,
 password=password,
 db=db,
 staging_fs_url=staging_fs_url,
 table=table)

# Read the data back into a Spark DataFrame
readDf = spark.read.load(
 # Spark format
 format=format,
 # Connector specific options
 host=host,
 user=user,
 password=password,
 db=db,
 table=table,
 staging_fs_url=staging_fs_url)

# Print the DataFrame contents
readDf.show()

File: examples/sparklyr/sparkapp.r
install.packages("curl", repo = "http://cran.us.r-project.org")
install.packages("sparklyr", repo = "http://cran.us.r-project.org")
library(sparklyr)

# Create a Spark config and disable Hive support to avoid errors
config <- spark_config()
config$sparklyr.jars.default <- "../../connector/target/scala-2.12/spark-vertica-connector-assembly-3.3.1.jar"
config$sparklyr.connect.enablehivesupport <- FALSE
config$sparklyr.appName <- "Vertica Spark Connector Sparklyr example"

print("Connecting to Spark.")

# Connect to the Spark cluster
sc <- spark_connect(master="spark://spark:7077", version = "3.1", config = config)

print("Connected to spark. Getting iris_tbl.")

# The Iris dataset comes with R and is used as test data
# Get the Iris data and store it in a Spark dataframe
iris_tbl <- sdf_copy_to(sc = sc, x = iris, overwrite = T)

print("Got iris_tbl. Writing to Vertica.")

# Write the Iris dataframe to the Vertica database
spark_write_source(iris_tbl, "com.vertica.spark.datasource.VerticaSource", "overwrite", list(
  "host" = "vertica",
  "user" = "dbadmin",
  "password" = "",
  "db" = "docker",
  "staging_fs_url" = "webhdfs://hdfs:50070/data/dirtest",
  "table" = "iris"
))

print("Wrote to Vertica. Reading from Vertica.")

# Read the Iris data back from the Vertica database into a Spark dataframe
result <- spark_read_source(sc = sc, name = "example", source = "com.vertica.spark.datasource.VerticaSource", options = list(
  "host" = "vertica",
  "user" = "dbadmin",
  "password" = "",
  "db" = "docker",
  "staging_fs_url" = "webhdfs://hdfs:50070/data/dirtest",
  "table" = "iris"
))

print("Finished reading.")

# Print the dataframe's contents
print(result)

# Cleanup Spark connection
spark_disconnect(sc)
